packages/preferences-api/src/common/preference-proxy.ts

```diff
@@ -55,16 +56,16 @@ export function createPreferenceProxy<T extends Configuration>(preferences: Pref
         get: (_, p: string) => {
             if (p in schema.properties) {
                 if (p in configuration) {
-                    const preference = preferences.get(p, configuration[p]);
+                    const preference = clone(preferences.get(p, configuration[p]));
```

Comment:
proxies are just convenient interfaces, functionality should be implemented in the lower level in `PreferenceService`